cpp/basix/finite-element.cpp

```diff
@@ -256,21 +256,81 @@ xt::xtensor<double, 3> basix::compute_expansion_coefficients(
   return xt::reshape_view(C, {num_dofs, vs, pdim});
 }
 //-----------------------------------------------------------------------------
+std::tuple<std::array<std::vector<xt::xtensor<double, 2>>, 4>,
+           std::array<std::vector<xt::xtensor<double, 3>>, 4>,
+           std::map<cell::type, xt::xtensor<double, 3>>>
+basix::make_discontinuous(
+    const std::array<std::vector<xt::xtensor<double, 2>>, 4>& x,
+    const std::array<std::vector<xt::xtensor<double, 3>>, 4>& M,
+    std::map<cell::type, xt::xtensor<double, 3>>& entity_transformations,
+    const int tdim, const int value_size)
+{
+  std::size_t npoints = 0;
+  std::size_t Mshape0 = 0;
+  for (int i = 0; i < 4; ++i)
+  {
+    for (std::size_t j = 0; j < x[i].size(); ++j)
+    {
+      npoints += x[i][j].shape(0);
+      Mshape0 += M[i][j].shape(0);
+    }
+  }
+
+  std::map<cell::type, xt::xtensor<double, 3>> entity_transformations_out;
+  std::array<std::vector<xt::xtensor<double, 3>>, 4> M_out;
+  std::array<std::vector<xt::xtensor<double, 2>>, 4> x_out;
+
+  xt::xtensor<double, 2> new_x
+      = xt::zeros<double>({npoints, static_cast<std::size_t>(tdim)});
+
+  xt::xtensor<double, 3> new_M = xt::zeros<double>(
+      {Mshape0, static_cast<std::size_t>(value_size), npoints});
+
+  int x_n = 0;
+  int M_n = 0;
+  for (int i = 0; i < 4; ++i)
+  {
+    for (std::size_t j = 0; j < x[i].size(); ++j)
+    {
+      xt::view(new_x, xt::range(x_n, x_n + x[i][j].shape(0)), xt::all())
+          .assign(x[i][j]);
+      xt::view(new_M, xt::range(M_n, M_n + M[i][j].shape(0)), xt::all(),
+               xt::range(x_n, x_n + x[i][j].shape(0)))
+          .assign(M[i][j]);
+      x_n += x[i][j].shape(0);
+      M_n += M[i][j].shape(0);
+    }
+  }
+
+  x_out[tdim].push_back(new_x);
+  M_out[tdim].push_back(new_M);
+
+  for (auto i = entity_transformations.begin();
```

Comment:
`auto it =` would more conventional. `i` is so commonly used as an integer index a reader could easily assume that it is.